docs/controller.md

```diff
@@ -0,0 +1,108 @@
+# Architecture Document for Controller Component
+
+## Description
+The Controller is the component giving meaning to the [pipeline](#structure) file given by the Monitor. It receives [pipelines](#pipeline), parse them into a set of [actions](#actions) and send these actions sequentially to the Scheduler, for each of these actions, the Scheduler **must** notify the Controller when a job has been scheduled and has been completed successfully or encountered an error, it can be done just by sending the logs from the Agent for instance. Thanks to these information, the Scheduler is able to provide information about a pipeline state to anyone (the Monitor or any other client).
+## Features
+- Users can send pipelines containing actions to execute. These actions are basically shell commands to execute. These pipelines are `yaml` files.
+- Users can track there actions by getting the logs from the Agent, the states of the job : `TODO`, `PENDING`, `DOING`, `COMPLETED`, `ERROR`. Refer to the sections [actions/states](#States).
+- The controller makes sure that each actions are executed in the right order (by design) and doesn't execute the next action if the previous one has failed.
+## Pipeline
```

Comment:
I appreciate that you define the `Pipeline` terminology, thanks. But lexicons and terminologies usually come first, before the terms they define are being used. So please make it the first section of this document/.

Reply:
Done, added a glossary section at line 3